frontend/components/panels/HighestBidCheckPanel.vue

```diff
@@ -0,0 +1,126 @@
+            auction, you receive your MKR back, only losing the transaction fees.
+        </TextBlock>
+
+        <div class="flex justify-between my-3">
+            Current Highest Bid
+            <span v-if="latestBid === null" class="text-gray-400"> There were no bids yet </span>
+            <FormatCurrency v-else :value="latestBid" currency="MKR" />
+        </div>
+
+        <div class="flex justify-end gap-5">
+            <BaseButton
+                class="w-full md:w-80"
+                :type="isUserLatestBidder ? 'secondary' : 'primary'"
+                :disabled="isBidding || isDisabled || !userWalletAddress || isLoading || !!error"
+                :is-loading="isBidding"
+                @click="$emit('restart')"
+            >
+                <span v-if="isBidding"> Bidding... </span>
+                <span v-else-if="auction"
+                    >Bid <FormatCurrency :value="bidAmount" currency="MKR" /> on
+                    <FormatCurrency :value="auction.receiveAmountDAI" currency="DAI"
+                /></span>
+                <span v-else-if="error">Unknown</span>
+                <span v-else>Loading...</span>
+            </BaseButton>
+        </div>
+    </BasePanel>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { SurplusAuction } from 'auctions-core/dist/src/types';
+import BigNumber from 'bignumber.js';
+import BasePanel from '../common/BasePanel.vue';
+import TextBlock from '../common/TextBlock.vue';
+import BaseButton from '../common/BaseButton.vue';
+import FormatCurrency from '../utils/FormatCurrency.vue';
+
+export default Vue.extend({
+    name: 'HighestBidCheckPanel',
+    components: { FormatCurrency, BaseButton, TextBlock, BasePanel },
+    props: {
+        auction: {
+            type: Object as Vue.PropType<SurplusAuction>,
+            default: null,
+        },
+        userWalletAddress: {
+            type: String,
+            default: null,
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+        bidAmount: {
+            type: Object as Vue.PropType<BigNumber>,
+            required: true,
+        },
+        isBidding: {
+            type: Boolean,
+            default: false,
+        },
+        isDisabled: {
+            type: Boolean,
+            default: false,
+        },
+        isLoading: {
+            type: Boolean,
+            default: false,
+        },
+        error: {
+            type: String,
+            default: null,
+        },
+    },
+    computed: {
+        latestBid() {
+            if (!this.auction || !this.auction.bidAmountMKR) {
+                return null;
+            }
+            return this.auction.bidAmountMKR;
+        },
+        isUserLatestBidder() {
+            if (!this.latestBid) {
+                return false;
+            }
+            return this.auction.receiverAddress === this.userWalletAddress;
+        },
```

Comment:
also nit 😄

```suggestion
        isUserLatestBidder(): boolean {
            return !!this.latestBid && this.auction.receiverAddress === this.userWalletAddress;
        },
```

Reply:
For both of these:
While they're both nice to haves, always type your computed properties, otherwise there's gonna be errors like
```
Property 'latestBid' does not exist on type 'CombinedVueInstance<Vue, unknown, unknown, unknown, Readonly<{ auction: SurplusAuction; userWalletAddress: string; isExplanationsShown: boolean; bidAmount: BigNumber; isBidding: boolean; isDisabled: boolean; isLoading: boolean; error: string; }>>'
```